recipes/tcl/8.6.10/conanfile.py

```diff
@@ -0,0 +1,222 @@
+from conans import ConanFile, AutoToolsBuildEnvironment, tools
+from conans.errors import ConanInvalidConfiguration
+from conans.util.env_reader import get_env
+import os
+import shutil
+import tempfile
+
+
+class TclConan(ConanFile):
+    name = "tcl"
+    description = "Tcl is a very powerful but easy to learn dynamic programming language."
+    topics = ("conan", "tcl", "scripting", "programming")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://tcl.tk"
+    license = "TCL"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "fPIC": [True, False],
+        "shared": [True, False]
+    }
+    default_options = {
+        "fPIC": True,
+        "shared": False,
+    }
+    _source_subfolder = "source_subfolder"
+    requires = ("zlib/1.2.11")
+
+    @property
+    def _is_mingw_windows(self):
+        return self.settings.os == "Windows" and self.settings.compiler == "gcc"
+
+    def configure(self):
+        if self.settings.compiler != "Visual Studio":
```

Comment:
After checking, the only C++ sources are in the `compat/zlib/contrib` folder. None of which are compiled by TCL, I removed it unconditionally.

Also, it is non-trivial to use `zlib` et. al as requirements, since there are rules to build them in TCL Makefiles. It would require quite a lot of patching.